deb/openmediavault/usr/share/php/openmediavault/system/storage/storagedevice.inc

```diff
@@ -33,23 +33,30 @@ class StorageDevice extends \OMV\System\BlockDevice
 	 * @return string The device model, otherwise an empty string.
 	 */
 	public function getModel() {
-		$filename = sprintf("/sys/block/%s/device/model",
-			$this->getDeviceName(TRUE));
-		if (file_exists($filename))
-			return trim(file_get_contents($filename));
-		return "";
+		if (FALSE === $this->hasUdevProperty("ID_MODEL")) {
+			$filename = sprintf("/sys/block/%s/device/model",
+				$this->getDeviceName(TRUE));
+			if (file_exists($filename))
+				return trim(file_get_contents($filename));
+			return "";
+		}
+		$property = $this->getUdevProperty("ID_MODEL");
+		return str_replace("_", " ", $property);  
 	}
 
 	/**
 	 * Get the device vendor.
 	 * @return string The device vendor, otherwise an empty string.
 	 */
 	public function getVendor() {
-		$filename = sprintf("/sys/block/%s/device/vendor",
-			$this->getDeviceName(TRUE));
-		if (file_exists($filename))
-			return trim(file_get_contents($filename));
-		return "";
+		if (FALSE === $this->hasUdevProperty("ID_VENDOR")) {
+			$filename = sprintf("/sys/block/%s/device/vendor",
+				$this->getDeviceName(TRUE));
+			if (file_exists($filename))
+				return trim(file_get_contents($filename));
+			return "";
+		}
+		$property = $this->getUdevProperty("ID_VENDOR");  		
```

Comment:
The return statement is missing.

Reply:
Ops! Copy / paste error when editing directly on github. Should be fixed now.